src/components/inputs/ListFilter/ListFilter.tsx

```diff
@@ -143,59 +148,73 @@ const ListFilter: React.FC<ListFilterProps> = ({
     onAddUserPreference(popUpUserPreference)
     setUserPreferencesIsDirty(false)
-  },[onAddUserPreference, popUpUserPreference])
+  }, [onAddUserPreference, popUpUserPreference])
 
-  const onListDeleteChangedLocal = useCallback((item: UserPreference) => {
-    if (confirm(localizedStrings.DeleteUserPreference) && item.filterName.toLowerCase() === selectedUserPreference.filterName.toLowerCase()) {
+  const onListDeleteChangedLocal = useCallback(
+    (item: UserPreference) => {
+      if (
+        confirm(localizedStrings.DeleteUserPreference) &&
+        item.filterName.toLowerCase() === selectedUserPreference.filterName.toLowerCase()
+      ) {
+        setPopUpUserPreference({
+          ...selectedUserPreference,
+          isDefault: false,
+          filterName: '',
+          implicit: false
+        })
+      }
+      onListDeleteChanged(item)
+    },
+    [localizedStrings.DeleteUserPreference, onListDeleteChanged, selectedUserPreference]
+  )
+
+  const onListImplicitChangedLocal = useCallback(
+    (item: UserPreference) => {
+      const newImplicitValue =
+        selectedUserPreference.filterName.toLowerCase() === item.filterName.toLowerCase()
+          ? !selectedUserPreference.implicit
+          : false
       setPopUpUserPreference({
         ...selectedUserPreference,
-        isDefault: false,
-        filterName: '',
-        implicit: false
+        implicit: newImplicitValue
       })
-    }
-    onListDeleteChanged(item)
-  },[localizedStrings.DeleteUserPreference, onListDeleteChanged, selectedUserPreference])
-
-  const onListImplicitChangedLocal = useCallback((item: UserPreference) => {
-    const newImplicitValue = selectedUserPreference.filterName.toLowerCase() === item.filterName.toLowerCase() ? !selectedUserPreference.implicit : false
-    setPopUpUserPreference({
-      ...selectedUserPreference,
-      implicit: newImplicitValue
-    })
-    onListImplicitChanged(item)
-  },[onListImplicitChanged, selectedUserPreference])
+      onListImplicitChanged(item)
+    },
+    [onListImplicitChanged, selectedUserPreference]
+  )
 
-  const hasChildren= useMemo(() => Boolean(children), [children])
-  const hasVisibleFilters= useMemo(() => Boolean(visibleFilters), [visibleFilters])
-  const hasVisibleFields= useMemo(() => Boolean(visibleFields), [visibleFields])
+  const hasChildren = useMemo(() => Boolean(children), [children])
+  const hasVisibleFilters = useMemo(() => Boolean(visibleFilters), [visibleFilters])
+  const hasVisibleFields = useMemo(() => Boolean(visibleFields), [visibleFields])
 
   return (
     <>
       <Toolbar sx={{ paddingLeft: 0, paddingRight: 0 }}>
-        <Grid container alignItems='center' spacing={2}>
+        <Grid container alignItems="center" spacing={2}>
           {(filters.fullTextFilter || filters.fullTextFilter === '') && (
-            <Grid item xs={12} md>
+            <Grid item xs={12} sm={12} md={visibleUserPreferences ? 3 : 5}>
```

Comment:
You can remove `sm={12}`. The sizes apply `greater than`, so having `xs={12}` and the same for `sm={12}` is redundant. This applies in multiple other places (for instance also specifying `md={12}`, which is triple redundant)

Reply:
I modified it to not be redundant in my last two commits